facade/dfs.go

```diff
@@ -554,6 +552,15 @@ func (f *Facade) Snapshot(ctx datastore.Context, serviceID, message string, tags
 			}
 		}
 	}
+	// Pause the services that need pausing in a batch
+	if _, err := scheduleServices(f, servicesToPause, ctx, tenantID, service.SVCPause, false); err != nil {
+		glog.Errorf("Could not pause services for snapshot: %s", err)
+		return "", err
+	}
+
+	defer scheduleServices(f, servicesToPause, ctx, tenantID, service.SVCRun, false)
+
+	// Wait for the paused services to reach the paused state (and other services to reach stopped)
```

Comment:
Note: Previously, the services were getting individual requests to pause and defer a return to previous state (which would always be run), this was changed to do a batch pause and defer a batch run.